types/shares.go

```diff
@@ -99,7 +99,8 @@ func split(rawData []byte, nid namespace.ID) []NamespacedShare {
 	rawData = rawData[MsgShareSize:]
 	for len(rawData) > 0 {
 		shareSizeOrLen := min(MsgShareSize, len(rawData))
-		rawShare := []byte(append(nid, rawData[:shareSizeOrLen]...))
+		rawShare := make([]byte, shareSizeOrLen)
+		copy(rawShare, rawData[:shareSizeOrLen])
```

Comment:
Not a suggestion to commit but what happens if this is used instead?
```suggestion
		rawShare := rawData[:shareSizeOrLen]
```